### PR TITLE
start of firewall waf functionality

### PR DESCRIPTION
**Pull request by bridgetlane**
Getting started on some of the WAF API functionality

### CODE REVIEW

**Comment by sethvargo** [comment]
Hey @bridgetlane 

This looks really good. Let me know when you're ready for a review or if you have any quesitons.

**Comment by bridgetlane** [comment]
Hi @sethvargo, I'd be happy to add testing but was not sure how you were generating your fixtures. Looks like they're using a personal service?

Otherwise this is ready for review, I can always add tests in a future PR if able 😄

File: fastly/waf.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	Included []FirewallData `mapstructure:"included"`
+
+	// Lists links in relation to your current page.
+	Links FirewallLinks `mapstructure:"links"`
+}
+
+// GetFirewallObjectsInput is the input needed to list firewall objects.
+type GetFirewallObjectsInput struct {
+	// Service is the ID of the service (required).
+	Service string
```
Comment (sethvargo) [inline fastly/waf.go]: I think all of the `*Input` structs need to declare the `form:"field,omitempty"` values

**Comment by bridgetlane** [comment]
@sethvargo pushed those changes

**Comment by sethvargo** [comment]
Hey @bridgetlane 

If you write the tests (you can copy the existing ones for format), I can run the suite against my account and generate the fixture!

**Comment by bridgetlane** [comment]
@sethvargo took a pass at the test file! Feedback appreciated 😄 

Did not commit the generated fixtures since obviously they were 401 unauthorized.

**Comment by sethvargo** [comment]
Hi @bridgetlane 

The WAF is still in private beta. I've requested access to test these changes.

**Comment by bridgetlane** [comment]
@sethvargo cool

**Comment by sethvargo** [comment]
Hi @bridgetlane 

I wanted to let you know that I haven't forgotten about this. I'm on a support thread with Fastly. In short, I don't think we are going to be able to get this supported in the near term. WAF is currently in very limited, invite-only preview. Many of the APIs listed (like [this one](https://docs.fastly.com/api/waf#waf_firewall_67e6cab552be64c18d2549cfc90ed378)) don't actually exist. There's currently no way to programmatically create/destroy a WAF, which is a requirement of this library for testing purposes. I'm going to think through some other options, but I don't want you to think I've forgotten about this.

**Comment by bridgetlane** [comment]
Hi @sethvargo, thanks for the update, I appreciate it.

**Pull request closed without merging**